### PR TITLE
chore(ci): use docker-compose for integration tests

### PR DESCRIPTION
**Pull request by jdrouet**
As mentioned in slack, we use bash script to start docker containers and podman containers.
Now that podman is compatible with docker-compose, we could use docker-compose to start and stop the containers.

Signed-off-by: Jérémie Drouet <jeremie.drouet@datadoghq.com>

### CODE REVIEW

File: scripts/setup_integration/docker-compose.redis.yml [hunk 1]
```diff
@@ -0,0 +1,6 @@
+services:
+  redis:
+    image: redis
```
Comment (spencergilbert) [inline scripts/setup_integration/docker-compose.redis.yml:3]: Unrelated to your change... I feel like we _may_ want to specify a tag?

**Review by spencergilbert — APPROVED** [review]

**Comment by binarylogic** [comment]
FWIW, we used Docker Compose in the past and removed it because it added a layer of indirection (#4549). We also went way overboard with how we used it, so what's going on here may not have the same issues.

**Comment by spencergilbert** [comment]
> #4549

Interesting... glancing over the PR they do look somewhat different. For integration tests, particularly ones with multiple docker containers being deployed it definitely feels less hacky using compose over bash scripting the docker commands. In this case where it's just a single container I'm more 🤷 - but see the argument for consistency.

**Comment by bruceg** [comment]
At least part of the motivation for the scripting was the desire to support both docker and podman. When the integration tests were first written, they were in fact set up using docker-compose, but podman lacked compatibility with that. There is now support for interpreting docker-compose files to produce podman pods, so it may be a good time to move _all_ support to the compose framework with a (hopefully trivial) shim to enable podman, which is our current testing default.

**Comment by spencergilbert** [comment]
> ...a (hopefully trivial) shim to enable podman, which is our current testing default.

As I understand, and last I tried, it _Just Works_ ™️

**Review by jszwedko — COMMENTED** [review]
I'm fan of this. I forgot, as Ben mentioned, that we used `docker-compose` before but it looks like we used it there as a sort of Makefile. I think using it for integration tests is clean.

File: scripts/setup_integration/docker-compose.redis.yml (hunk 1 above)
Comment (jdrouet) [reply scripts/setup_integration/docker-compose.redis.yml:3]: we should, yes!

**Comment by jdrouet** [comment]
do we run the integration tests with podman on the CI?

**Comment by jdrouet** [comment]
If you want 😄 
If we want to support it, shouldn't we test it on the CI as well?

**Comment by spencergilbert** [comment]
> If you want 😄 If we want to support it, shouldn't we test it on the CI as well?

I don't think we need to run our CI suite twice for each, imho

**Comment by jdrouet** [comment]
> I don't think we need to run our CI suite twice for each, imho

Well, if we don't test it on the CI, it should probably not be a "feature" and we should probably not add complexity about it in the test scripts 🤔

**Comment by spencergilbert** [comment]
I'll leave that decision to the higher-ups 🤷

**Comment by jdrouet** [comment]
@binarylogic what do you think? Is @bruceg 's argument enough?

**Comment by spencergilbert** [comment]
I'm personally OK with this, and _overall_ think we should decide on truly supporting one of Docker or Podman (while accepting contributions/easy to maintain or fix issues for the other). Since at large the org seems to prefer Docker that seems the natural choice.

In terms of `docker-compose` and Podman: https://fedoramagazine.org/use-docker-compose-with-podman-to-orchestrate-containers-on-fedora/ && https://github.com/containers/podman-compose

**Comment by jdrouet** [comment]
This PR is replaced by #10466 which pushes further this idea of using docker-compose

**Pull request closed without merging**